direct/nn/unet/unet_engine.py

```diff
@@ -35,15 +78,208 @@ def __init__(
             **models,
         )
 
-    def forward_function(self, data: Dict[str, Any]) -> Tuple[torch.Tensor, None]:
-        output_image = self.model(
-            masked_kspace=data["masked_kspace"],
-            sensitivity_map=(
-                data["sensitivity_map"] if self.cfg.model.image_initialization == "sense" else None  # type: ignore
-            ),
+    def forward_function(self, data: dict[str, Any]) -> tuple[torch.Tensor, None]:
+        """Forward function for :class:`Unet2dEngine`.
+
+        Parameters
+        ----------
+        data : dict[str, Any]
+            Input data dictionary containing the following keys: "masked_kspace" and "sensitivity_map"
+            if image initialization is "sense".
+
+        Returns
+        -------
+        tuple[torch.Tensor, None]
+            Prediction of image and None for k-space.
+        """
+
+        sensitity_map = (
+            data["sensitivity_map"] if self.cfg.model.image_initialization == "sense" else None  # type: ignore
         )
+
+        output_image = self.model(masked_kspace=data["masked_kspace"], sensitivity_map=sensitity_map)
         output_image = T.modulus(output_image)
 
         output_kspace = None
 
         return output_image, output_kspace
+
+
+class Unet2dSSLEngine(SSLMRIModelEngine):
+    """SSL Unet2d Model Engine.
+
+    Parameters
+    ----------
```

Comment:
Perhaps a reference where it is used?

Reply:
I used it in the JSSL paper. Referenced. Also done with E2EVarNet